source/App.cpp

```diff
@@ -1327,32 +1195,21 @@ void App::onUserInput(UserInput* ui) {
 					if (sess->canFire()) {
 						fired = true;
 						sess->countClick();														// Count clicks
-						Array<shared_ptr<Entity>> dontHit = { m_hitDecal };
-						dontHit.append(m_currentMissDecals);
+						Array<shared_ptr<Entity>> dontHit;
 						dontHit.append(m_explosions);
-						for (auto projectile : m_projectileArray) { dontHit.append(projectile.entity); }
```

Comment:
I notice the `dontHit` array no longer includes projectiles, hit decals and miss decals. Is there a reason for this?

Reply:
This now gets handle at [line 133 of `Weapon::simulateProjectiles()`](https://github.com/NVlabs/abstract-fps/pull/114/files#diff-bd4b3d1e04cccc48e9bbdeb92de4b6c8R133) we only need to add the things that the weapon doesn't own/isn't passed here.

We pass in the target array, so for now it's just explosions that remain belonging to the `App`.